coding-challenges.js

```diff
@@ -1,60 +1,369 @@
+/// - - - - - - - - - - - - - - - -
 
 
 
 
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, -59, 8, 24]
-// Expected output: [-59, 109]
 
 
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
 
+// --------------------4)   Create a function that takes in a string and returns a string with every other letter capitalized.
 
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
 
+var testString1 = "albatross"   // Expected output: "aLbAtRoSs"
+var testString2 = "jabberwocky" // Expected output: "jAbBeRwOcKy"
 
 
-var testString1 = "albatross"
-// Expected output: "aLbAtRoSs"
-var testString2 = "jabberwocky"
-// Expected output: "jAbBeRwOcKy"
 
+const capitalizeEveryOtherLetter = strIn => {
+    return Array.from( strIn ).map( (c,i) => { if (i%2===1) return c.toUpperCase(); return c } ).join("")
+}
 
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
 
 
+/// - - - - - - - - - - - - - - - -
 
+log("Capitalize Every Other Letter",
 
+        `${testString1}     >>   ${capitalizeEveryOtherLetter( testString1 )}`,
+        `${testString2}   >>   ${capitalizeEveryOtherLetter( testString2 )}`
+    )
+
+/// - - - - - - - - - - - - - - - -
+
+
+
+
+
+
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
+
+// --------------------5)   Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
 
 var arr1 = [3, 7, 10, 5, 4, 3, 3]
 var arr2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+//   -->   console.log(...arr1)
+
+
+
+const joinArraysAndRemoveDuplicates = (a1, a2, a3, i=0) => {
+
+    // Set Up
+    //      >   Makes sure there is at least 1 array to use.
+    //      >   A double equal sign is used so 'null' or 'undefined' can be used interchangeably
+    if (a3 == undefined){  a3 = a1.concat(a2)  }
+
+    // Getting the indexes of the first and last instance of an item.
+    let [ a3_FIRST, a3_LAST ]  =  [ a3.indexOf( a3[i] ), a3.lastIndexOf( a3[i] ) ]
+
+    // Removing the last duplicate of an item
+    if ( a3_FIRST !== a3_LAST ){
+        a3.splice(a3_LAST,1)
+        joinArraysAndRemoveDuplicates( null, null, a3, i )
+    }
+    else if ( i < a3.length ){
+        joinArraysAndRemoveDuplicates( null, null, a3, ++i )
+    }
+
+
+    return a3
+}
```

Comment:
this is another one that could use some more commenting. it works great, you just want to be able to effectively communicate what your process was in these solutions.